pango/src/glyph_item_iter.rs

```diff
@@ -1,63 +1,186 @@
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::pango_glyph_item_iter_get_type()) }
+    }
+}
+
+impl<'item, 'text> GlyphItemIter<'item, 'text> {
+    #[doc(alias = "pango_glyph_item_iter_next_cluster")]
+    pub fn next_cluster(&mut self) -> bool {
+        unsafe {
+            from_glib(ffi::pango_glyph_item_iter_next_cluster(
+                self.to_glib_none_mut().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "pango_glyph_item_iter_prev_cluster")]
+    pub fn prev_cluster(&mut self) -> bool {
+        unsafe {
+            from_glib(ffi::pango_glyph_item_iter_prev_cluster(
+                self.to_glib_none_mut().0,
+            ))
+        }
+    }
     #[doc(alias = "pango_glyph_item_iter_init_end")]
-    pub fn new_end(glyph_item: &GlyphItem, text: &str) -> Result<Self, glib::BoolError> {
+    pub fn new_end(
+        glyph_item: &'item GlyphItem,
+        text: impl AsRef<GStr> + 'text,
+    ) -> Result<Self, glib::BoolError> {
         unsafe {
             let mut iter = mem::MaybeUninit::zeroed();
             let res: bool = from_glib(ffi::pango_glyph_item_iter_init_end(
                 iter.as_mut_ptr(),
                 mut_override(glyph_item.to_glib_none().0),
-                text.to_glib_none().0,
+                text.as_ref().as_ptr(),
             ));
 
             if res {
-                Ok(from_glib_none(&iter.assume_init() as *const _))
+                Ok(Self {
+                    inner: iter.assume_init(),
+                    item: PhantomData,
+                    text: PhantomData,
+                })
             } else {
                 Err(glib::bool_error!("Failed to create glyph item iter"))
             }
         }
     }
 
     #[doc(alias = "pango_glyph_item_iter_init_start")]
-    pub fn new_start(glyph_item: &GlyphItem, text: &str) -> Result<Self, glib::BoolError> {
+    pub fn new_start(
+        glyph_item: &'item GlyphItem,
+        text: impl AsRef<GStr> + 'text,
+    ) -> Result<Self, glib::BoolError> {
         unsafe {
             let mut iter = mem::MaybeUninit::zeroed();
             let res: bool = from_glib(ffi::pango_glyph_item_iter_init_start(
                 iter.as_mut_ptr(),
                 mut_override(glyph_item.to_glib_none().0),
-                text.to_glib_none().0,
+                text.as_ref().as_ptr(),
             ));
 
             if res {
-                Ok(from_glib_none(&iter.assume_init() as *const _))
+                Ok(Self {
+                    inner: iter.assume_init(),
+                    item: PhantomData,
+                    text: PhantomData,
+                })
             } else {
                 Err(glib::bool_error!("Failed to create glyph item iter"))
             }
         }
     }
 
-    pub fn start_char(&self) -> i32 {
-        unsafe { (*self.to_glib_none().0).start_char }
+    #[inline]
+    pub fn glyph_item(&self) -> &'item GlyphItem {
+        unsafe { mem::transmute(&self.inner.glyph_item) }
```

Comment:
```suggestion
        unsafe { &*(&self.inner.glyph_item as *const _ as *const GlyphItem) }
```

Is slightly safer (it checks for more things than transmute)